### PR TITLE
GEM-39 | Implementation of "POST /internal/user-details" endpoint

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/external/client/RestAttachmentStoreClient.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+class RestAttachmentStoreClient(
+    @Qualifier("AttachmentStoreRestTemplate") val restTemplate: RestTemplate,
+    val attachmentStoreProperties: AttachmentStoreProperties,
+) : AttachmentStoreClient {
+    @Retry(name = "attachmentStore")
+    override fun getDefaultUserAttachmentId(): String {
+        return try {
+            restTemplate.exchange(
+                resolveDefaultAttachmentIdAddress(),
+                GET,
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/client/RestAttachmentStoreClient.kt]: for creation of new attachemnt, IMHO i think is more accureste to use POST

File: src/main/kotlin/pl/edu/agh/gem/external/dto/UserDetailsCreationRequest.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    fun toDomain(attachmentId: String) =
+        UserDetails(
+            id = userId,
+            username = username,
+            firstName = null,
+            lastName = null,
+            phoneNumber = null,
+            bankAccountNumber = null,
+            attachmentId = attachmentId,
+            preferredPaymentMethod = NONE,
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/UserDetailsCreationRequest.kt]: could you move default to the data class ?

File: src/integration/kotlin/pl/edu/agh/gem/integration/controller/UserDetailsControllerIT.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        should("create user details") {
+            // given
+            val userDetailsRequest = createUserDetailRequest()
+            stubDefaultAttachmentIdUrl(createDefaultAttachmentResponse())
+
+            // when
+            val response = service.createUserDetails(userDetailsRequest)
+
+            // then
+            response shouldHaveHttpStatus CREATED
```
Comment (lawos98) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controller/UserDetailsControllerIT.kt:25]: could you also check if we have saved in repository ?

Comment (pwgacek) [reply src/integration/kotlin/pl/edu/agh/gem/integration/controller/UserDetailsControllerIT.kt:25]: how? At the moment repository has only save method. Should i add it in pr which adds findById to repository?

Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/controller/UserDetailsControllerIT.kt:25]: you could inject mongooperation, or add that in that PR ( with that method )

**Review by lawos98 — APPROVED** [review]

**Pull request merged**